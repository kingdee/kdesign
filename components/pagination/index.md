---
category: Components
type: 基础
order: 3
title: Pagination
subtitle: 分页

---

分页是一种分隔列表数据的组件

## 使用场景
列表数据量较多时。

## API

| 参数               | 说明                                                                          | 类型                                         | 默认值 | 版本 |
|------------------|-----------------------------------------------------------------------------|--------------------------------------------| --- | --- |
| bordered         | 是否加边框                                                                       | boolean                                    | false |  |
| current          | 当前页码                                                                        | number                                     | - | 1.0.0 |
| defaultCurrent   | 默认当前页码                                                                      | number                                     | 1 | 1.0.0 |
| defaultPageSize  | 默认每页条码                                                                      | number                                     | 10 | 1.0.0 |
| disabled         | 禁用分页                                                                        | boolean                                    | - | 1.0.0 |
| dropdownProps    | `pageSize` 切换器属性                                                            | [DropdownProps](/components/dropdown/#API) | - | 1.0.0 |
| hideOnSinglePage | 只有一页时是否隐藏分页组件                                                               | boolean                                    | false | 1.0.0 |
| pageSize         | 每页条目数                                                                       | number                                     | - | 1.0.0 |
| pageSizeOptions  | `pageSize` 切换器的下拉列表                                                         | string\[]                                  | \[`10`, `20`, `50`, `100`] | 1.0.0 |
| pageType         | 分页类型，基础（basic)、单据(bill)、简化(simple)、较少(less)和精细(nicety)可选，默认是基础(basic)       | `basic` \                                  | `bill` \| `simple` \| `less` \| `nicety` | `basic` | 1.0.0 |
| showTotal        | 是否显示总计，默认为false, 当pageType为`basic`时，默认为`page`, 当pageType为`nicety`时，默认为`row` | boolean \                                  | `page` \| `row` \| `all` | false | 1.0.0 |
| showQuickJumper  | 是否可以快速跳转至某页，默认为true, 当pageType为`bill`时，默认为false                             | boolean                                    | true | 1.0.0 |
| showSizeChanger  | 是否展示 `pageSize` 切换器，当 `total` 大于 50 时默认为 true                               | boolean                                    | - | 1.0.0 |
| showTitle        | 是否显示原生 tooltip 页码                                                           | boolean                                    | true | 1.0.0 |
| total            | 数据总量                                                                        | number                                     | 0 | 1.0.0 |
| onChange         | 页码改变的回调，参数是改变后的页码及每页条目数                                                     | function(page, pageSize)                   | - | 1.0.0 |
| onShowSizeChange | pageSize 变化后的回调                                                             | function(current, size)                    | - | 1.0.0 |
| icons            | 替换图标                                                                        | IIcon                                  | {} | 1.0.0 |

### IIcon

| key      | 说明   |
|----------|------|
| first    | 第一页  |
| last     | 最后一页 |
| prev     | 前一页  |
| next     | 后一页  |
| jumpPrev | 向前多页 |
| jumpNext | 向后多页 |
| down     | 下拉   |


## Design Token

| 分类 | 组件token | 全局token | 默认值 |
| --- | --- | --- | --- |
| color | --kd-c-pagination-text-color | --kd-g-color-text-secondary | #666 |
|  | --kd-c-pagination-color-text-hover | --kd-g-color-theme | #5582f3 |
|  | --kd-c-pagination-button-color | --kd-g-color-text-secondary | #666 |
|  | --kd-c-pagination-color-disabled | --kd-g-color-disabled | #b2b2b2 |
|  | --kd-c-pagination-color-border | --kd-g-color-border-strong | #d9d9d9 |
|  | --kd-c-pagination-color-border-hover | --kd-g-color-theme-7 | rgb(55, 92, 202) |
|  | --kd-c-pagination-color-border-disabled | --kd-g-color-border-strong | #d9d9d9 |
|  | --kd-c-pagination-button-color-background-disabled | --kd-g-color-background-contain-disabled | #f5f5f5 |
| font | --kd-c-pagination-font-size | --kd-g-font-size-small | 12px |
|  | --kd-c-pagination-icon-font-size | --kd-g-font-size-large | 16px |
| sizing | --kd-c-pagination-sizing-square | - | 24px |
|  | --kd-c-pagination-sizing-border-width | - | 1px |
| spacing | --kd-c-pagination-button-spacing | - | 4px |
|  | --kd-c-pagination-dropdown-item-spacing-padding-horizontal | - | 24px |
