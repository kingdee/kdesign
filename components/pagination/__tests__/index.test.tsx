import React from 'react'
import { render, mount } from 'enzyme'
import mountTest from '../../../tests/shared/mountTest'
import ConfigProvider from '../../config-provider/index'
import Pagination from '../index'
import Icon from '../../icon'
import { PageTypes } from '../pagination'
const defaultPaginationProps = {
  onChange: jest.fn(),
  onShowSizeChange: jest.fn(),
}
describe('Pagination', () => {
  // 1.mount test
  describe('1.mount test', () => {
    mountTest(Pagination)
    PageTypes.forEach((type) => {
      mountTest(() => <Pagination pageType={type} defaultCurrent={1} total={50} />)
    })
  })

  // 2.render test
  describe('2.render test', () => {
    it('renders correctly', () => {
      expect(render(<Pagination defaultCurrent={1} total={50} />)).toMatchSnapshot()
      PageTypes.forEach((type) => {
        expect(render(<Pagination pageType={type} defaultCurrent={1} total={50} />)).toMatchSnapshot()
      })
    })
  })

  // 3.render no child without errors
  describe('3.render no child without errors', () => {
    it('render no child without errors', () => {
      expect(mount(<Pagination defaultCurrent={1} total={50}></Pagination>)).toMatchSnapshot()
    })
  })

  // 4. render null or undefined without errors
  describe('4. render null or undefined without errors', () => {
    it('render null or undefined without errors', () => {
      expect(
        mount(
          <Pagination defaultCurrent={1} total={50}>
            {null}
            {undefined}
          </Pagination>,
        ),
      ).toMatchSnapshot()
    })
  })
  // 5. displayName
  describe('5. displayName', () => {
    it('should have displayName static property', () => {
      const wrapper = mount(<Pagination defaultCurrent={1} total={50} />)
      expect((wrapper.type() as any).displayName).toBe('Pagination')
    })
  })
  // 6.class state
  describe('6. class state', () => {
    it('should class use right', () => {
      // default
      const DefaultPagination = mount(<Pagination defaultCurrent={1} total={50} />)
      expect(DefaultPagination.find('.kd-pagination')).toHaveClassName('.kd-pagination')
    })
  })
  // 7.component interaction(event)
  describe('7.component interaction(event)', () => {
    it('should not clickable when button is loading', () => {
      const onClick = jest.fn()
      const wrapper = mount(<Pagination defaultCurrent={3} total={500} disabled />)
      wrapper.simulate('click')
      expect(onClick).not.toHaveBeenCalledWith()
    })
  })
  // 8.config provider
  describe('8.config provider', () => {
    it('should config use config provider', () => {
      const paginationConfig = {
        compDefaultProps: {
          Pagination: {
            current: 3,
          },
        },
        localeConfig: {
          locale: 'zh-CN',
          localeData: {
            'Pagination.confirm': 'GO',
          },
        },
      }
      const wrapper = mount(
        <ConfigProvider value={paginationConfig}>
          <Pagination pageType="nicety" total={50} />
        </ConfigProvider>,
      )
      expect(wrapper.find(`.kd-pagination-pages-item.active`).at(0).text()).toEqual('3')
      expect(wrapper.find(`.kd-pagination-jumper-button`).at(0).text()).toEqual('GO')
    })
  })
  // 9.API test
  describe('9.api test', () => {
    it('api test', () => {
      // border
      const defaultwrapper = mount(
        <Pagination bordered pageType="simple" defaultCurrent={2} total={50} showSizeChanger />,
      )
      expect(defaultwrapper.find(`.kd-pagination-action`)).toHaveClassName(`.bordered`)

      // current
      const defalutPagination = mount(<Pagination current={3} total={50} />)
      expect(defalutPagination.find(`.kd-pagination-current-input`).at(0).prop('value')).toEqual(3)

      const billPagination = mount(<Pagination pageType="bill" current={3} total={50} showQuickJumper />)
      expect(billPagination.find(`.kd-pagination-current-input`).at(0).prop('value')).toEqual(3)

      const nicetyPagination = mount(<Pagination pageType="nicety" current={3} total={50} />)
      expect(nicetyPagination.find(`.kd-pagination-pages-item.active`).at(0).prop('title')).toEqual('3')

      const lessPagination = mount(<Pagination pageType="less" current={3} total={50} />)
      expect(lessPagination.find(`.kd-pagination-pages-item.active`).at(0).prop('title')).toEqual('3')
      const simplePagination = mount(<Pagination bordered pageType="simple" current={2} total={50} />)
      expect(simplePagination.find(`.kd-pagination-current-input`).at(0).prop('value')).toEqual(2)
      // defaultCurrent
      expect(defalutPagination.find(`.kd-pagination-current-input`).at(0).prop('value')).toEqual(3)

      const billPaginationTwo = mount(<Pagination pageType="bill" defaultCurrent={3} total={50} showQuickJumper />)
      expect(billPaginationTwo.find(`.kd-pagination-current-input`).at(0).prop('value')).toEqual(3)

      const nicetyPaginationTwo = mount(<Pagination pageType="nicety" defaultCurrent={3} total={50} />)
      expect(nicetyPaginationTwo.find(`.kd-pagination-pages-item.active`).at(0).prop('title')).toEqual('3')

      const lessPaginationTwo = mount(<Pagination pageType="less" defaultCurrent={3} total={50} />)
      expect(lessPaginationTwo.find(`.kd-pagination-pages-item.active`).at(0).prop('title')).toEqual('3')
      const simplePaginationTwo = mount(<Pagination bordered pageType="simple" defaultCurrent={2} total={50} />)
      expect(simplePaginationTwo.find(`.kd-pagination-current-input`).at(0).prop('value')).toEqual(2)

      // defaultPageSize
      const wrapper = mount(<Pagination defaultPageSize={20} defaultCurrent={1} total={50} showSizeChanger />)
      expect(wrapper.find(`.kd-dropdown-trigger`).find('span').at(0).text()).toEqual(`20 条/页`)

      // dropdownProps
      // hideOnSinglePage
      const hidePagination = mount(<Pagination defaultCurrent={1} total={10} showSizeChanger hideOnSinglePage />)
      expect(hidePagination.exists('.kd-dropdown')).toBeFalsy()

      // pageSize
      const pageSizePagination = mount(<Pagination pageSize={30} defaultCurrent={1} total={100} />)
      expect(pageSizePagination.find(`.kd-dropdown-trigger`).find('span').at(0).text()).toEqual(`30 条/页`)

      // pageSizeOptions
      // const pageSizeOptionsPagination = mount(
      //   <Pagination showSizeChanger pageSizeOptions={['10', '20', '30', '40']} defaultCurrent={1} total={100} />,
      // )
      // expect(pageSizeOptionsPagination.find('.kd-dropdown-menu-item').at(0)).toHaveText('苹果')
      // expect(pageSizeOptionsPagination.find('.kd-dropdown-menu-item').at(1)).toHaveText('橘子')
      // expect(pageSizeOptionsPagination.find('.kd-dropdown-menu-item').at(2)).toHaveText('橘子')
      // expect(pageSizeOptionsPagination.find('.kd-dropdown-menu-item').at(3)).toHaveText('橘子')

      // showTotal
      const showTotalPagePagination = mount(<Pagination showTotal={'page'} defaultCurrent={1} total={500} />)
      expect(showTotalPagePagination.find('.kd-pagination-total').at(0)).toHaveText('共 50 页')

      const showTotalRowPagination = mount(<Pagination showTotal={'row'} defaultCurrent={1} total={500} />)
      expect(showTotalRowPagination.find('.kd-pagination-total').at(0)).toHaveText('共 500 条')

      const showTotalAllPagination = mount(<Pagination showTotal={'all'} defaultCurrent={1} total={500} />)
      expect(showTotalAllPagination.find('.kd-pagination-total').at(0)).toHaveText('共 50 页 500条')
      // showQuickJumper

      const showQuickJumperPagination = mount(
        <Pagination showQuickJumper pageType="nicety" defaultCurrent={1} total={50} />,
      )
      expect(showQuickJumperPagination.exists('.kd-pagination-jumper')).toBeTruthy()
      // showSizeChanger
      const showSizeChangerPagination = mount(
        <Pagination showSizeChanger={false} pageType="nicety" defaultCurrent={1} total={49} />,
      )
      expect(showSizeChangerPagination.exists('.kd-pagination-options')).toBeFalsy()

      const showSizeChangerPagination2 = mount(<Pagination pageType="nicety" defaultCurrent={1} total={129} />)
      expect(showSizeChangerPagination2.exists('.kd-pagination-options')).toBeTruthy()

      // showTitle
      // total
      const totalRowPagination = mount(<Pagination showTotal={'row'} defaultCurrent={1} total={1000} />)
      expect(totalRowPagination.find('.kd-pagination-total').at(0)).toHaveText('共 1000 条')
      // onChange
      const onChangePagination = mount(
        <Pagination pageType="nicety" pageSize={20} defaultCurrent={1} total={50} {...defaultPaginationProps} />,
      )
      onChangePagination.find('.kd-pagination-pages-item').at(3).find('button').simulate('click')

      expect(defaultPaginationProps.onChange).toHaveBeenCalled()
      // onShowSizeChange

      // icons
      const iconsPagination = mount(<Pagination pageType="bill" total={50} bordered />)
      iconsPagination.setProps({ icons: { first: <Icon type="arrow-up" /> } })
      expect(iconsPagination.find('.kdicon-arrow-up').length).toBe(1)
    })
  })
})
