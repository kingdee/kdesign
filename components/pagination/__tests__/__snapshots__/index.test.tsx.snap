// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination 2.render test renders correctly 1`] = `
<div
  class="kd-pagination"
>
  <span
    class="kd-pagination-total"
  >
    共 5 页
  </span>
  <span
    class="kd-pagination-current"
  >
    <span>
      第 
    </span>
    <input
      class="kd-pagination-current-input"
      type="text"
      value="1"
    />
    <span>
       页
    </span>
  </span>
  <ul
    class="kd-pagination-action"
  >
    <li
      class="kd-pagination-action-item"
      title="第一页"
    >
      <button
        disabled=""
      >
        <i
          class="kdicon kdicon-first"
        />
      </button>
    </li>
    <li
      class="kd-pagination-action-item"
      title="上一页"
    >
      <button
        disabled=""
      >
        <i
          class="kdicon kdicon-arrow-left"
        />
      </button>
    </li>
    <li
      class="kd-pagination-action-item"
      title="下一页"
    >
      <button>
        <i
          class="kdicon kdicon-arrow-right"
        />
      </button>
    </li>
    <li
      class="kd-pagination-action-item"
      title="最后一页"
    >
      <button>
        <i
          class="kdicon kdicon-last"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`Pagination 2.render test renders correctly 2`] = `
<div
  class="kd-pagination"
>
  <span
    class="kd-pagination-total"
  >
    共 5 页
  </span>
  <span
    class="kd-pagination-current"
  >
    <span>
      第 
    </span>
    <input
      class="kd-pagination-current-input"
      type="text"
      value="1"
    />
    <span>
       页
    </span>
  </span>
  <ul
    class="kd-pagination-action"
  >
    <li
      class="kd-pagination-action-item"
      title="第一页"
    >
      <button
        disabled=""
      >
        <i
          class="kdicon kdicon-first"
        />
      </button>
    </li>
    <li
      class="kd-pagination-action-item"
      title="上一页"
    >
      <button
        disabled=""
      >
        <i
          class="kdicon kdicon-arrow-left"
        />
      </button>
    </li>
    <li
      class="kd-pagination-action-item"
      title="下一页"
    >
      <button>
        <i
          class="kdicon kdicon-arrow-right"
        />
      </button>
    </li>
    <li
      class="kd-pagination-action-item"
      title="最后一页"
    >
      <button>
        <i
          class="kdicon kdicon-last"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`Pagination 2.render test renders correctly 3`] = `
<div
  class="kd-pagination"
>
  <ul
    class="kd-pagination-action"
  >
    <li
      class="kd-pagination-action-item"
      title="第一页"
    >
      <button
        disabled=""
      >
        <i
          class="kdicon kdicon-first"
        />
      </button>
    </li>
    <li
      class="kd-pagination-action-item"
      title="上一页"
    >
      <button
        disabled=""
      >
        <i
          class="kdicon kdicon-arrow-left"
        />
      </button>
    </li>
    <li
      class="kd-pagination-action-item"
      title="下一页"
    >
      <button>
        <i
          class="kdicon kdicon-arrow-right"
        />
      </button>
    </li>
    <li
      class="kd-pagination-action-item"
      title="最后一页"
    >
      <button>
        <i
          class="kdicon kdicon-last"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`Pagination 2.render test renders correctly 4`] = `
<div
  class="kd-pagination simple"
>
  <ul
    class="kd-pagination-action"
  >
    <li
      class="kd-pagination-action-item"
      title="上一页"
    >
      <button
        disabled=""
      >
        <i
          class="kdicon kdicon-arrow-left"
        />
      </button>
    </li>
    <li
      class="kd-pagination-action-item"
      title="1/5"
    >
      <span
        class="kd-pagination-current"
      >
        <input
          class="kd-pagination-current-input"
          type="text"
          value="1"
        />
      </span>
      / 
      <span
        class="kd-pagination-total"
      >
        5
      </span>
    </li>
    <li
      class="kd-pagination-action-item"
      title="下一页"
    >
      <button>
        <i
          class="kdicon kdicon-arrow-right"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`Pagination 2.render test renders correctly 5`] = `
<div
  class="kd-pagination less"
>
  <ul
    class="kd-pagination-pages"
  >
    <li
      class="kd-pagination-pages-item"
      title="上一页"
    >
      <button
        disabled=""
      >
        <i
          class="kdicon kdicon-arrow-left"
        />
      </button>
    </li>
    <li
      class="kd-pagination-pages-item active"
      title="1"
    >
      <button>
        1
      </button>
    </li>
    <li
      class="kd-pagination-pages-item"
      title="2"
    >
      <button>
        2
      </button>
    </li>
    <li
      class="kd-pagination-pages-item"
      title="3"
    >
      <button>
        3
      </button>
    </li>
    <li
      class="kd-pagination-pages-item"
      title="4"
    >
      <button>
        4
      </button>
    </li>
    <li
      class="kd-pagination-pages-item"
      title="5"
    >
      <button>
        5
      </button>
    </li>
    <li
      class="kd-pagination-pages-item"
      title="下一页"
    >
      <button>
        <i
          class="kdicon kdicon-arrow-right"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`Pagination 2.render test renders correctly 6`] = `
<div
  class="kd-pagination nicety"
>
  <span
    class="kd-pagination-total"
  >
    共 50 条
  </span>
  <ul
    class="kd-pagination-pages"
  >
    <li
      class="kd-pagination-pages-item"
      title="上一页"
    >
      <button
        disabled=""
      >
        <i
          class="kdicon kdicon-arrow-left"
        />
      </button>
    </li>
    <li
      class="kd-pagination-pages-item active"
      title="1"
    >
      <button>
        1
      </button>
    </li>
    <li
      class="kd-pagination-pages-item"
      title="2"
    >
      <button>
        2
      </button>
    </li>
    <li
      class="kd-pagination-pages-item"
      title="3"
    >
      <button>
        3
      </button>
    </li>
    <li
      class="kd-pagination-pages-item"
      title="4"
    >
      <button>
        4
      </button>
    </li>
    <li
      class="kd-pagination-pages-item"
      title="5"
    >
      <button>
        5
      </button>
    </li>
    <li
      class="kd-pagination-pages-item"
      title="下一页"
    >
      <button>
        <i
          class="kdicon kdicon-arrow-right"
        />
      </button>
    </li>
  </ul>
  <div
    class="kd-pagination-jumper"
  >
    <input
      class="kd-pagination-jumper-input"
      type="text"
      value=""
    />
    <button
      class="kd-pagination-jumper-button"
    >
      GO
    </button>
  </div>
</div>
`;

exports[`Pagination 3.render no child without errors render no child without errors 1`] = `
<Pagination
  defaultCurrent={1}
  total={50}
>
  <div
    className="kd-pagination"
  >
    <span
      className="kd-pagination-total"
    >
      共 5 页
    </span>
    <span
      className="kd-pagination-current"
    >
      <span
        key="0"
      >
        第 
      </span>
      <input
        className="kd-pagination-current-input"
        key="1"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyUp={[Function]}
        type="text"
        value={1}
      />
      <span
        key="2"
      >
         页
      </span>
    </span>
    <ul
      className="kd-pagination-action"
    >
      <li
        className="kd-pagination-action-item"
        title="第一页"
      >
        <button
          disabled={true}
          onClick={[Function]}
        >
          <Icon
            type="first"
          >
            <i
              className="kdicon kdicon-first"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={Object {}}
            />
          </Icon>
        </button>
      </li>
      <li
        className="kd-pagination-action-item"
        title="上一页"
      >
        <button
          disabled={true}
          onClick={[Function]}
        >
          <Icon
            type="arrow-left"
          >
            <i
              className="kdicon kdicon-arrow-left"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={Object {}}
            />
          </Icon>
        </button>
      </li>
      <li
        className="kd-pagination-action-item"
        title="下一页"
      >
        <button
          onClick={[Function]}
        >
          <Icon
            type="arrow-right"
          >
            <i
              className="kdicon kdicon-arrow-right"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={Object {}}
            />
          </Icon>
        </button>
      </li>
      <li
        className="kd-pagination-action-item"
        title="最后一页"
      >
        <button
          onClick={[Function]}
        >
          <Icon
            type="last"
          >
            <i
              className="kdicon kdicon-last"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={Object {}}
            />
          </Icon>
        </button>
      </li>
    </ul>
  </div>
</Pagination>
`;

exports[`Pagination 4. render null or undefined without errors render null or undefined without errors 1`] = `
<Pagination
  defaultCurrent={1}
  total={50}
>
  <div
    className="kd-pagination"
  >
    <span
      className="kd-pagination-total"
    >
      共 5 页
    </span>
    <span
      className="kd-pagination-current"
    >
      <span
        key="0"
      >
        第 
      </span>
      <input
        className="kd-pagination-current-input"
        key="1"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyUp={[Function]}
        type="text"
        value={1}
      />
      <span
        key="2"
      >
         页
      </span>
    </span>
    <ul
      className="kd-pagination-action"
    >
      <li
        className="kd-pagination-action-item"
        title="第一页"
      >
        <button
          disabled={true}
          onClick={[Function]}
        >
          <Icon
            type="first"
          >
            <i
              className="kdicon kdicon-first"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={Object {}}
            />
          </Icon>
        </button>
      </li>
      <li
        className="kd-pagination-action-item"
        title="上一页"
      >
        <button
          disabled={true}
          onClick={[Function]}
        >
          <Icon
            type="arrow-left"
          >
            <i
              className="kdicon kdicon-arrow-left"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={Object {}}
            />
          </Icon>
        </button>
      </li>
      <li
        className="kd-pagination-action-item"
        title="下一页"
      >
        <button
          onClick={[Function]}
        >
          <Icon
            type="arrow-right"
          >
            <i
              className="kdicon kdicon-arrow-right"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={Object {}}
            />
          </Icon>
        </button>
      </li>
      <li
        className="kd-pagination-action-item"
        title="最后一页"
      >
        <button
          onClick={[Function]}
        >
          <Icon
            type="last"
          >
            <i
              className="kdicon kdicon-last"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={Object {}}
            />
          </Icon>
        </button>
      </li>
    </ul>
  </div>
</Pagination>
`;
